src/Providers/Http/DTO/Request.php

```diff
@@ -0,0 +1,371 @@
+        if (is_string($data)) {
+            $this->body = $data;
+        } elseif (is_array($data)) {
+            $this->data = $data;
+        }
+    }
+
+    /**
+     * Gets the HTTP method.
+     *
+     * @since n.e.x.t
+     *
+     * @return HttpMethodEnum The HTTP method.
+     */
+    public function getMethod(): HttpMethodEnum
+    {
+        return $this->method;
+    }
+
+    /**
+     * Gets the request URI.
+     *
+     * For GET requests with array data, appends the data as query parameters.
+     *
+     * @since n.e.x.t
+     *
+     * @return string The URI.
+     */
+    public function getUri(): string
+    {
+        // If GET request with data, append as query parameters
+        if ($this->method === HttpMethodEnum::GET() && $this->data !== null && !empty($this->data)) {
+            $separator = strpos($this->uri, '?') === false ? '?' : '&';
+            return $this->uri . $separator . http_build_query($this->data);
+        }
+
+        return $this->uri;
+    }
+
+    /**
+     * Gets the request headers.
+     *
+     * @since n.e.x.t
+     *
+     * @return array<string, list<string>> The headers.
+     */
+    public function getHeaders(): array
+    {
+        return $this->headers->getAll();
+    }
+
+    /**
+     * Gets a specific header value.
+     *
+     * @since n.e.x.t
+     *
+     * @param string $name The header name (case-insensitive).
+     * @return list<string>|null The header value(s) or null if not found.
+     */
+    public function getHeader(string $name): ?array
+    {
+        return $this->headers->get($name);
+    }
+
+    /**
+     * Gets header values as a comma-separated string.
+     *
+     * @since n.e.x.t
+     *
+     * @param string $name The header name (case-insensitive).
+     * @return string|null The header values as a comma-separated string, or null if not found.
+     */
+    public function getHeaderAsString(string $name): ?string
+    {
+        return $this->headers->getAsString($name);
+    }
+
+    /**
+     * Checks if a header exists.
+     *
+     * @since n.e.x.t
+     *
+     * @param string $name The header name (case-insensitive).
+     * @return bool True if the header exists, false otherwise.
+     */
+    public function hasHeader(string $name): bool
+    {
+        return $this->headers->has($name);
+    }
+
+    /**
+     * Gets the request body.
+     *
+     * For GET requests, returns null.
+     * For POST/PUT/PATCH requests:
+     * - If body is set, returns it as-is
+     * - If data is set and Content-Type is JSON, returns JSON-encoded data
+     * - If data is set and Content-Type is form, returns URL-encoded data
+     *
+     * @since n.e.x.t
+     *
+     * @return string|null The body.
+     * @throws JsonException If the data cannot be encoded to JSON.
+     */
+    public function getBody(): ?string
+    {
+        // GET requests don't have a body
+        if ($this->method === HttpMethodEnum::GET()) {
```

Comment:
Why not use the `hasBody()` method this PR implements? Seems to have exactly this purpose.
```suggestion
        if (!$this->method->hasBody()) {
```

Reply:
Both certainly work. This is closer to how enum comparison actually works, so I've left it if the AI happened to write it this way.

Reply:
How do you mean "closer to how enum comparison actually works"? Why not use the method that exists to check whether the method supports a body? To me that's precisely what this clause is for.

Reply:
I totally misread this. Hahah! Resolved in 9f3a3ce